### PR TITLE
Allow PHP8 and fix issue with MongoQuerySerializer

### PR DESCRIPTION
**Pull request by starred-gijs**
Not sure if you are planning to support PHP8, but that would be great!

I tried it and got some errors with MongoQuerySerializer:
`method_exists(): Argument #1 ($object_or_class) must be of type object|string, array given`

Did not encounter any other issues, but I also do not use all features. Would it be possible to run the travis build on PHP8 too?

### CODE REVIEW

File: src/DataCollector/MongoQuerySerializer.php [hunk 1]
```diff
@@ -70,7 +74,7 @@ public static function prepareItemData($item)
             return $item->bsonSerialize();
         }
 
-        if (\is_array($item) || \is_object($item)) {
+        if (\is_object($item)) {
```
Comment (Jean85) [inline src/DataCollector/MongoQuerySerializer.php:77]: If `|| \is_object($item)` is checked at line 57, this if is futile then.

**Review by Jean85 — CHANGES_REQUESTED** [review]

**Comment by Jean85** [comment]
Thank you for your contribution! Could you also add a test for this bug you found?

File: src/DataCollector/MongoQuerySerializer.php (hunk 1 above)
Comment (starred-gijs) [reply src/DataCollector/MongoQuerySerializer.php:77]: Yes I made a mistake while re-applying my changes in the fork. Updated line 57

**Comment by starred-gijs** [comment]
> Thank you for your contribution! Could you also add a test for this bug you found?

I added tests for scalar values, but breaks on 
https://github.com/facile-it/mongodb-bundle/blob/master/tests/Unit/DataCollector/MongoQuerySerializerTest.php#L30

Because the cloned value is actually equal. Not sure why those tests are there?

File: .idea/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (ilario-pierbattista) [inline .idea/.gitignore:1]: .idea files should not be here

File: .idea/modules.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilario-pierbattista) [inline .idea/modules.xml:1]: .idea files should not be here

File: .idea/mongodb-bundle.iml [hunk 4]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilario-pierbattista) [inline .idea/mongodb-bundle.iml:1]: .idea files should not be here

File: .idea/php.xml [hunk 5]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilario-pierbattista) [inline .idea/php.xml:1]: .idea files should not be here

File: .idea/phpunit.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilario-pierbattista) [inline .idea/phpunit.xml:1]: .idea files should not be here

File: .idea/vcs.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilario-pierbattista) [inline .idea/vcs.xml:1]: .idea files should not be here

**Review by ilario-pierbattista — CHANGES_REQUESTED** [review]
Hi @starred-gijs, thanks for contributing. 
Your .idea folder leaked into this PR. Feel free to add a specific rule in the general .gitconfig file

**Comment by ilario-pierbattista** [comment]
> Did not encounter any other issues, but I also do not use all features. Would it be possible to run the travis build on PHP8 too?

@starred-gijs we're having some trouble with the new billing plans of travis.
I'm going to migrate all the builds to github actions asap

**Comment by ilario-pierbattista** [comment]
@starred-gijs Thanks to @Jean85 the migration was super fast! 
You can rebase your branch with master and add a php-8 stage in the ci matrix

File: composer.json [hunk 8]
```diff
@@ -19,18 +19,18 @@
...
     },
     "require-dev": {
-        "matthiasnoback/symfony-dependency-injection-test": "^3",
+        "matthiasnoback/symfony-dependency-injection-test": "^3 || ^4",
         "symfony/web-profiler-bundle": "^3.4 || ^4.3",
         "symfony/console": "^3.4 || ^4.3",
         "phpunit/phpunit": "^7.0",
         "symfony/phpunit-bridge": "^4.2",
-        "facile-it/facile-coding-standard": "^0.4.0",
+        "facile-it/facile-coding-standard": "^0.4.0 || dev-master",
```
Comment (Jean85) [inline composer.json:33]: https://github.com/facile-it/facile-coding-standard/releases/tag/0.4.1
```suggestion
        "facile-it/facile-coding-standard": "^0.4.1",
```

**Comment by starred-gijs** [comment]
Sorry I tried to get all the issues resolved, but it still needs some work... Wanna help out?

**Review by ilario-pierbattista — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 9]
```diff
@@ -63,6 +63,12 @@ jobs:
           - php: 7.4
             mongo-ext: 1.9.0
             mongo-img: percona/percona-server-mongodb:4.4
+          - php: 8.0
+            mongo-ext: 1.6.0
```
Comment (ilario-pierbattista) [inline .github/workflows/ci.yaml:67]: https://docs.mongodb.com/drivers/php/#language-compatibility

According to this only ext 1.9 seems to be compatible with php8

**Comment by ilario-pierbattista** [comment]
@starred-gijs these are mostly deprecation notices from symfony 5.1.
Can you try forcing symfony the 5.0 version and see what happen?

**Comment by starred-gijs** [comment]
> Can you try forcing symfony the 5.0 version and see what happen?

How do I do that?

**Comment by ilario-pierbattista** [comment]
Add `symfony: '5.0.*'` to the item you added in the ci matrix. 
The first one of them does a similar thing

**Comment by ilario-pierbattista** [comment]
@starred-gijs can you rebase again with master? #107 is now merged

**Comment by Jean85** [comment]
Enabling PHPUnit 9 doesn't seem so straightforward unfortunately :cry: and deprecations are getting in the way too.
I'll try to address those with two separate PRs..

**Comment by starred-gijs** [comment]
What I could do it revert the PHP8/ PHPUnit changes, and just get the changes for `MongoQuerySerializer` in?

**Comment by Jean85** [comment]
@starred-gijs that would be ok, but I would suggest to just port them to a separate PR. In that way we would split this apart, and obtain small mergeable pieces :+1: 

I've already started with #111, one would be the fix with the serializer and we would need an other one with PHPUnit 9. Waht do you want to tackle?

**Comment by starred-gijs** [comment]
I will create a new PR for only the serializer fixes

**Comment by Jean85** [comment]
Closing this since we're splitting it:
 * #112 fixes the issue with the query serializer
 * #113 updates to PHPUnit 8
 * #111 supports Symfony 5 properly

**Pull request closed without merging**